src/Categories/Category/Finite/Fin/Instance/HGraph.agda

```diff
@@ -0,0 +1,146 @@
+{-# OPTIONS --without-K --safe #-}
+
+module Categories.Category.Finite.Fin.Instance.HGraph where
+
+open import Data.Nat.Base using (ℕ)
+open import Data.Fin.Base using (Fin)
+open import Data.Fin.Patterns
+open import Relation.Binary.PropositionalEquality.Core using (_≡_ ; refl)
+
+open import Categories.Category.Finite.Fin
+open import Categories.Category
+open import Categories.Functor using (Functor; _∘F_) renaming (id to idF)
+open import Categories.Functor.Presheaf
+
+open import Categories.NaturalTransformation using (ntHelper)
+open import Categories.NaturalTransformation.NaturalIsomorphism using (_≃_)
+open import Categories.Adjoint.Equivalence using (⊣Equivalence)
+open import Categories.Adjoint.TwoSided using (_⊣⊢_; withZig)
+import Categories.Morphism as Mor
+
+
+private
+  variable
+    a b c d : Fin 2
+
+-- The diagram is the following:
+-- 0 ---n--> 1
+-- with n arrows between 0 and 1
+
+HyperGraphShape : ℕ → FinCatShape
+HyperGraphShape n = shapeHelper record
+  { size      = 2
+  ; ∣_⇒_∣     = morph
+  ; id        = id
+  ; _∘_       = _∘_
+  ; assoc     = assoc
+  ; identityˡ = identityˡ
+  ; identityʳ = identityʳ
+  }
+  where morph : Fin 2 → Fin 2 → ℕ
+        morph 0F 0F = 1
+        morph 0F 1F = n
+        morph 1F 0F = 0
+        morph 1F 1F = 1
+        
+        id : Fin (morph a a)
+        id {0F} = 0F -- 0F is the only inhabitant of Fin 1
+        id {1F} = 0F 
+
+        _∘_ : ∀ {a b c} → Fin (morph b c) → Fin (morph a b) → Fin (morph a c)
+        _∘_ {0F} {0F} {0F} 1₀ 1₀' = 1₀
+        _∘_ {0F} {0F} {1F} g 1₀   = g
+        _∘_ {1F} {1F} {1F} 1₁ 1₁' = 1₁
+        _∘_ {0F} {1F} {1F} 1₁ f   = f
+   
+        assoc : ∀ {f : Fin (morph a b)} {g : Fin (morph b c)} {h : Fin (morph c d)} →
```

Comment:
using private `variables` is great, unless you need to explicitly given them a value (as you do here). Then it is much better to actually give them explicitly (with different names, just to be sure) because of *ordering*. Between large versions of Agda, the algorithm that choose the ordering of insertion of implicit variable changes, causing all such code to break.  So here it is best to make them explicit.